src/main/java/hudson/plugins/ec2/WindowsData.java

```diff
@@ -22,7 +27,18 @@ public WindowsData(
             boolean useHTTPS,
             String bootDelay,
             boolean specifyPassword,
-            boolean allowSelfSignedCertificate) {
+            boolean allowSelfSignedCertificate)
+            throws Descriptor.FormException {
+        try {
+            FIPS140Utils.ensureNoPasswordLeak(useHTTPS, password);
+        } catch (IllegalArgumentException e) {
+            throw new Descriptor.FormException(e, "password");
+        }
+        try {
+            FIPS140Utils.ensureNoSelfSignedCertificate(allowSelfSignedCertificate);
+        } catch (IllegalArgumentException e) {
+            throw new Descriptor.FormException(e, "allowSelfSignedCertificate");
+        }
```

Comment:
There is no check on the password length here if a password is in used.  Also should the checks only take place if `specifyPassword` is true?

Reply:
Also: 93c5a3c9ba06a35f3d5c95d6fd37431b7b25332a